CAIPs/caip-222.md

```diff
@@ -0,0 +1,249 @@
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_authenticate",
+  "params": {
+      "cacaov": string,
+      "type": string,
+      "chainId": string,
+      "domain": string,
+      "aud": string,
+      "version": string,
+      "nonce": string,
+      "iat": string,
+      "exp": string, // optional
+      "nbf": string, // optional
+      "statement": string, // optional
+      "requestId": string, // optional
+      "resources": string[], // optional
+  }
+}
+```
+
+The JSON-RPC method is labelled as `wallet_authenticate` and expects the following parameters:
+
+- cacaov - Cacao version number
+- type - Cacao header message type
+- chainId - [CAIP-2][]-defined `chainId` to identify specific chain or network.
+- domain - [RFC 4501][rfc 4501] `dnsauthority` that is requesting the signing.
+- aud - [RFC 3986][rfc 3986] URI referring to the resource that is the subject of the signing.
+- version - Current version of the message.
+- nonce - Randomized token to prevent signature replay attacks.
+- iat - [RFC 3339][rfc 3339] `date-time` that indicates the issuance time.
+- exp (optional) - [RFC 3339][rfc 3339] `date-time` that indicates when the signed authentication message is no longer valid.
+- nbf (optional) - [RFC 3339][rfc 3339] `date-time` that indicates when the signed authentication message starts being valid.
+- statement (optional) - Human-readable ASCII assertion that the user will sign. It _MUST NOT_ contain `\n`.
+- requestId (optional) - System-specific identifier used to uniquely refer to the authentication request.
+- resources (optional) - List of information or references to information the user wishes to have resolved as part of the authentication by the relying party; express as [RFC 3986][rfc 3986] URIs and separated by `\n`.
+
+### Response
+
+The wallet will prompt the user with a dedicated UI to display the app requesting the authentication and allow the user to select which blockchain account to sign with.
+
+#### Success
+
+If approved, the walllet will return a signed CACAO with the selected blockchain account.
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "h": {
+      "t": string,
+    },
+    "p": {
+      "iss": string,
+      "aud": string,
+      "exp": string,
+      "iat": string,
+      "nbf": string,
+      "nonce": string,
+      "domain": string,
+      "version": string,
+      "requestId": string,
+      "resources": string[],
+      "statement": string,
+    },
+    "s": {
+      "t": string,
+      "s": string
+    }
+  }
+}
+```
+
+All payload parameters MUST match the same parameters as the ones provided in the request and the following will be added:
+
+- h - header object according to [CAIP-74][caip-74]
+  - t - cacao message type (matching `type` in request)
+- p - payload object according to [CAIP-74][caip-74]
+  - iss - [did:pkh][did:pkh] method identifying [CAIP-10][caip-10] blockchain account
```

Comment:
I like the idea of using did:pkh here, but does it address chain forks? I think forks could present security issues here

Reply:
That's a deeper problem than did:pkh or even CAIP-10

For example, EIP-155 doesn't handle chain forks really well 

That means that one of the chains must change it's chainId